shared/config.go

```diff
@@ -63,6 +68,119 @@ type Config struct {
 	RequestCompression RequestCompressionFunc
 	// NodeHealthStoreConfig controls node health tracking logic
 	NodeHealthStoreConfig nodeshealth.NodeHealthStoreConfig
+	// KeyRouteAffinity configures which operations should use routing optimization heuristics
+	KeyRouteAffinity KeyRouteAffinityConfig
+}
+
+// OpType specifies the type of operations that should use routing optimization
+type OpType int
+
+const (
+	// KeyRouteAffinityNone disables routing optimization for all operations
+	KeyRouteAffinityNone OpType = iota
+	// KeyRouteAffinityWrite enables routing optimization for all write operations (Put, Update, Delete)
+	KeyRouteAffinityWrite
+	// KeyRouteAffinityAll enables routing optimization for all operations including reads
+	KeyRouteAffinityAll
+)
+
+// KeyRouteAffinityConfig holds configuration for routing optimization heuristics
+type KeyRouteAffinityConfig struct {
+	Type           OpType
+	QueryPlanSeed  int64
```

Comment:
This is not part of this feature, please remove it, put it back to master config